webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/ThirdPartyTMSSmartling.java

```diff
@@ -91,11 +137,106 @@ public void createImageToTextUnitMappings(String projectId, List<ThirdPartyImage
         smartlingClient.createBindings(bindings, projectId);
     }
 
-    Pattern getFilePattern(String repositoryName) {
-        return Pattern.compile(repositoryName + "/(\\d+)_(singular|plural)_source.xml");
+    @Override
+    public void push(Repository repository,
+                     String projectId,
+                     String pluralSeparator,
+                     String skipTextUnitsWithPattern,
+                     String skipAssetsWithPathPattern,
+                     List<String> optionList) {
+
+        lastPushResult = new ArrayList<>();
+        List<String> localeTags = ImmutableList.of("en");
+        SmartlingOptions options = SmartlingOptions.parseList(optionList);
+        AndroidStringDocumentMapper mapper = new AndroidStringDocumentMapper(pluralSeparator, null);
+
+        AtomicInteger batchCount = new AtomicInteger(0);
```

Comment:
to associate the batch number:  https://guava.dev/releases/23.0/api/docs/com/google/common/collect/Streams.html#mapWithIndex-java.util.stream.Stream-com.google.common.collect.Streams.FunctionWithIndex- ?

Reply:
💯